// plugin-welcome/src/api/models/ControllersPositionassingment.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersPositionassingment
 */
export interface ControllersPositionassingment {
    /**
     * 
     * @type {string}
     * @memberof ControllersPositionassingment
     */
    dayStart?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersPositionassingment
     */
    formdata?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersPositionassingment
     */
    formdepartment?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersPositionassingment
     */
    formposition?: number;
}

export function ControllersPositionassingmentFromJSON(json: any): ControllersPositionassingment {
    return ControllersPositionassingmentFromJSONTyped(json, false);
}

export function ControllersPositionassingmentFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersPositionassingment {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'dayStart': !exists(json, 'dayStart') ? undefined : json['dayStart'],
        'formdata': !exists(json, 'formdata') ? undefined : json['formdata'],
        'formdepartment': !exists(json, 'formdepartment') ? undefined : json['formdepartment'],
        'formposition': !exists(json, 'formposition') ? undefined : json['formposition'],
    };
}

export function ControllersPositionassingmentToJSON(value?: ControllersPositionassingment | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'dayStart': value.dayStart,
        'formdata': value.formdata,
        'formdepartment': value.formdepartment,
        'formposition': value.formposition,
    };
}
