// plugin-welcome/src/api/models/EntPositionassingmentEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDepartment,
    EntDepartmentFromJSON,
    EntDepartmentFromJSONTyped,
    EntDepartmentToJSON,
    EntPhysician,
    EntPhysicianFromJSON,
    EntPhysicianFromJSONTyped,
    EntPhysicianToJSON,
    EntPosition,
    EntPositionFromJSON,
    EntPositionFromJSONTyped,
    EntPositionToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPositionassingmentEdges
 */
export interface EntPositionassingmentEdges {
    /**
     * 
     * @type {EntDepartment}
     * @memberof EntPositionassingmentEdges
     */
    department?: EntDepartment;
    /**
     * 
     * @type {EntPosition}
     * @memberof EntPositionassingmentEdges
     */
    position?: EntPosition;
    /**
     * 
     * @type {EntPhysician}
     * @memberof EntPositionassingmentEdges
     */
    user?: EntPhysician;
}

export function EntPositionassingmentEdgesFromJSON(json: any): EntPositionassingmentEdges {
    return EntPositionassingmentEdgesFromJSONTyped(json, false);
}

export function EntPositionassingmentEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPositionassingmentEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'department': !exists(json, 'department') ? undefined : EntDepartmentFromJSON(json['department']),
        'position': !exists(json, 'position') ? undefined : EntPositionFromJSON(json['position']),
        'user': !exists(json, 'user') ? undefined : EntPhysicianFromJSON(json['user']),
    };
}

export function EntPositionassingmentEdgesToJSON(value?: EntPositionassingmentEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'department': EntDepartmentToJSON(value.department),
        'position': EntPositionToJSON(value.position),
        'user': EntPhysicianToJSON(value.user),
    };
}
