// plugin-welcome/src/api/models/EntPhysician.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPhysicianEdges,
    EntPhysicianEdgesFromJSON,
    EntPhysicianEdgesFromJSONTyped,
    EntPhysicianEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPhysician
 */
export interface EntPhysician {
    /**
     * EMAIL holds the value of the "EMAIL" field.
     * @type {string}
     * @memberof EntPhysician
     */
    eMAIL?: string;
    /**
     * NAME holds the value of the "NAME" field.
     * @type {string}
     * @memberof EntPhysician
     */
    nAME?: string;
    /**
     * 
     * @type {EntPhysicianEdges}
     * @memberof EntPhysician
     */
    edges?: EntPhysicianEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPhysician
     */
    id?: number;
}

export function EntPhysicianFromJSON(json: any): EntPhysician {
    return EntPhysicianFromJSONTyped(json, false);
}

export function EntPhysicianFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPhysician {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'eMAIL': !exists(json, 'EMAIL') ? undefined : json['EMAIL'],
        'nAME': !exists(json, 'NAME') ? undefined : json['NAME'],
        'edges': !exists(json, 'edges') ? undefined : EntPhysicianEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntPhysicianToJSON(value?: EntPhysician | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'EMAIL': value.eMAIL,
        'NAME': value.nAME,
        'edges': EntPhysicianEdgesToJSON(value.edges),
        'id': value.id,
    };
}
